crates/project_model/src/workspace.rs

```diff
@@ -234,6 +234,23 @@ impl ProjectWorkspace {
                                 .and_then(|it| it.out_dir.clone()),
                         );
 
+                        // In case target's path is manually set in Cargo.toml to be
+                        // outside the package root, add its parent as an extra include.
+                        // An example of this situation would look like this:
+                        //
+                        // ```toml
+                        // [lib]
+                        // path = "../../src/lib.rs"
+                        // ```
+                        let extra_targets = cargo[pkg]
+                            .targets
+                            .iter()
+                            .filter(|&&tgt| cargo[tgt].kind == TargetKind::Lib)
+                            .filter_map(|&tgt| cargo[tgt].root.parent())
+                            .map(|tgt| tgt.normalize().to_path_buf())
```

Comment:
FYI, path.starts_with(pkg_root.clone()) always returned true, unless we normalized.